force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,105 @@
+public class OpportunityTriggerHandler implements ITriggerHandler {
+	private static Boolean triggerDisabled = false;
+
+	public void disableTrigger() {
+		triggerDisabled = true;
+	}
+	public void enableTrigger() {
+		triggerDisabled = false;
+	}
+
+	public Boolean isTriggerEnabled() {
+		if (triggerDisabled) {
+			return false;
+		} else {
+			return true;
+		}
+	}
+
+	public void afterUpdate(
+		Map<Id, SObject> newItemsMap,
+		Map<Id, SObject> oldItemsMap,
+		List<SObject> newItems,
+		List<SObject> oldItems
+	) {
+		List<Task> taskList = new List<Task>();
```

Comment:
it's better to move your logic inside afterUpdate, beforeUpdate, etc to separate method in this TriggerHandler. 
Because logic will grow and we can't keep all in one method for possibility to read this and understand what is going on.
So we can make some method integrateWithPartner(), handleRelatedWorkBreakdownStructure(), whatever your logic really does